src/encoder/mod.rs

```diff
@@ -451,7 +445,7 @@ pub struct DirectoryEncoder<'a, W: 'a + Write + Seek> {
 impl<'a, W: 'a + Write + Seek> DirectoryEncoder<'a, W> {
     fn new(writer: &'a mut TiffWriter<W>) -> TiffResult<DirectoryEncoder<'a, W>> {
         // the previous word is the IFD offset position
-        let ifd_pointer_pos = writer.offset() - mem::size_of::<u32>() as u64;
+        let ifd_pointer_pos = writer.offset() - u64::try_from(mem::size_of::<u32>())?;
```

Comment:
The `u64::try_from(_)` is a constant expression. I doesn't make a lot of sense to error at runtime here. I'd like to diagnose this at compile time but am not sure if it is pretty.

Reply:
Yeah, don't know what I was thinking with this one -- it should just be `as u64` since `4` will obviously cast fine to `u64`, as would any real world `size_of`. This might be related to some generics I was doing for BigTiff branch, but if so I should change it there.

Eventually `unwrap` will be a const fn, which would allow a simple unwrap on `try_from` to throw at compile time or otherwise be optimized out in cases like this.